### PR TITLE
Rebase nov 2023 into master branch

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - Tested locally for bucket mounting, unmounting, listing, stat, read operations
2. Unit tests - tested locally, all passed.
3. Integration tests - ran as part of presubmit.

### CODE REVIEW

File: perfmetrics/scripts/run_load_test_and_fetch_metrics.sh [hunk 1]
```diff
@@ -30,21 +30,27 @@ git checkout c5d8ce3fc736210ded83b126c71e3225c7ffd7c9 && \
 ./configure && make && sudo make install
 
 cd "${KOKORO_ARTIFACTS_DIR}/github/gcsfuse/perfmetrics/scripts"
+
```
Comment (gargnitingoogle) [inline perfmetrics/scripts/run_load_test_and_fetch_metrics.sh:33]: There were some conflicts here with the changes in @Tulsishah's [commit](https://github.com/GoogleCloudPlatform/gcsfuse/commit/1a5db44f520d4854d5418342ca597a85f654f5c8#diff-ff86a10afec13e1928d1cf8ff4f5c5c5fe7aed9794cd54e6931d9509e1b96643) in master and @ashmeenkaur's [commit in nov_2023_release](https://github.com/GoogleCloudPlatform/gcsfuse/commit/42c828c1e8278d84211aa28908eb9cb3403ec9fd). Please check this file carefully.

Comment (raj-prince) [reply perfmetrics/scripts/run_load_test_and_fetch_metrics.sh:33]: Looks good to me.

Comment (gargnitingoogle) [reply perfmetrics/scripts/run_load_test_and_fetch_metrics.sh:33]: @raj-prince thanks.

File: perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh [hunk 2]
```diff
@@ -62,6 +62,7 @@ function execute_perf_test() {
 if [[ "$perfTestStr" == *"$EXECUTE_PERF_TEST_LABEL"* ]];
 then
  # Executing perf tests for master branch
+ git reset --hard
```
Comment (raj-prince) [inline perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh:65]: Please revert before check-in.

**Comment by gargnitingoogle** [comment]
presubmit tests (both perf and integration) passed. Build log [here](https://fusion2.corp.google.com/ci/kokoro/prod:gcsfuse%2Fgcp_ubuntu%2Fpresubmits%2Fperf_tests%2Fpresubmit/activity/e8d9aa7d-4e44-4235-9515-ced990dec2f4/summary) .

Perf results
```
+--------+------------+--------------+------------+--------------+--------------+
| Branch | File Size  |   Read BW    |  Write BW  | RandRead BW  | RandWrite BW |
+--------+------------+--------------+------------+--------------+--------------+
| Master |  0.25MiB   | 413.27MiB/s  | 1.07MiB/s  |  31.23MiB/s  |  1.15MiB/s   |
|   PR   |  0.25MiB   | 430.16MiB/s  | 1.12MiB/s  |  33.16MiB/s  |  1.27MiB/s   |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
| Master | 48.828MiB  | 4751.57MiB/s | 77.96MiB/s | 1042.48MiB/s |  77.12MiB/s  |
|   PR   | 48.828MiB  | 4598.99MiB/s | 80.92MiB/s | 1126.9MiB/s  |  79.6MiB/s   |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
| Master | 976.562MiB | 4608.75MiB/s | 32.56MiB/s | 528.26MiB/s  |  33.15MiB/s  |
|   PR   | 976.562MiB | 4600.75MiB/s | 28.65MiB/s | 533.01MiB/s  |  31.18MiB/s  |
|        |            |              |            |              |              |
|        |            |              |            |              |              |
+--------+------------+--------------+------------+--------------+--------------+
```

File: perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh (hunk 2 above)
Comment (gargnitingoogle) [reply perfmetrics/scripts/presubmit_test/pr_perf_test/build.sh]: Done. Removed.

**Comment by gargnitingoogle** [comment]
@raj-prince @sethiay @ashmeenkaur  please approve.

**Review by ashmeenkaur — APPROVED** [review]

**Review by sethiay — APPROVED** [review]
Relying on Prince's and Ashmeen's review.

**Pull request merged**